packages/ses-ava/src/ses-ava-test.js

```diff
@@ -46,45 +66,88 @@ const logErrorFirst = (func, args, name, logger = defaultLogger) => {
   'failing',
   'serial',
   'only',
-  'skip',
 ];
 
 /**
- * @param {TesterFunc} testerFunc
- * @param {Logger} [logger]
- * @returns {TesterFunc} Not yet frozen!
+ * @callback BaseImplFunc
+ * This is the function that invariably starts `t => {`.
+ * Ava's types call this `Implementation`, but that's just too confusing.
+ * @param {Assertions} t
+ * @returns {unknown}
+ *
+ * @typedef {BaseImplFunc | Object} ImplFunc
+ * @property {(...unknown) => string} [title]
+ *
+ * @callback TesterFunc
+ * @param {string} title
+ * @param {ImplFunc} [implFunc]
+ * @returns {void}
  */
-const wrapTester = (testerFunc, logger = defaultLogger) => {
+
+/**
+ * @template {TesterFunc} T
+ * @param {T} testerFunc
+ * @param {Logger} [logError]
+ * @returns {T} Not yet frozen!
+ */
+const augmentLogging = (testerFunc, logError) => {
+  const testerFuncName = `ava ${testerFunc.name || 'test'}`;
   /** @type {TesterFunc} */
-  const testerWrapper = (title, implFunc, ...otherArgs) => {
-    /** @type {ImplFunc} */
-    const testFuncWrapper = t => {
-      harden(t);
-      return logErrorFirst(implFunc, [t, ...otherArgs], 'ava test', logger);
+  const augmented = (...args) => {
+    // Align with ava argument parsing.
+    // https://github.com/avajs/ava/blob/c74934853db1d387c46ed1f953970c777feed6a0/lib/parse-test-args.js
+    const maybeTitle = typeof args[0] === 'string' ? [args.shift()] : [];
+    const implFuncOrObj = args.shift();
+    const wrapImplFunc = fn => {
+      const wrappedFunc = t => {
+        harden(t);
+        return logErrorFirst(fn, [t, ...args], testerFuncName, logError);
+      };
+      if (fn.title) {
+        wrappedFunc.title = fn.title;
+      }
+      return wrappedFunc;
     };
-    if (implFunc && implFunc.title) {
-      testFuncWrapper.title = implFunc.title;
+    let implArg;
+    if (typeof implFuncOrObj === 'function') {
+      // Handle common cases like `test(title, t => { ... }, ...)`.
+      implArg = wrapImplFunc(implFuncOrObj);
+    } else if (typeof implFuncOrObj === 'object' && implFuncOrObj) {
+      // Handle cases like `test(title, test.macro(...), ...)`.
+      // Note that this will need updating if a future version of ava adds an alternative to `exec`.
+      implArg = freeze({
+        ...implFuncOrObj,
+        exec: wrapImplFunc(implFuncOrObj.exec),
+      });
+    } else {
+      // Let ava handle this bad input.
+      implArg = implFuncOrObj;
     }
-    return testerFunc(title, testFuncWrapper, ...otherArgs);
+    // @ts-expect-error these spreads are acceptable
+    return testerFunc(...maybeTitle, implArg, ...args);
   };
-  return testerWrapper;
+  // re-use other properties (e.g. `.always`)
+  // https://github.com/endojs/endo/issues/647#issuecomment-809010961
+  Object.assign(augmented, testerFunc);
+  // @ts-expect-error cast
+  return augmented;
 };
 
 /**
- * The ava `test` function takes a callback argument of the form
+ * The Ava `test` function takes a callback argument of the form
```

Comment:
+1 to proper noun. Same two lines below.

Reply:
Except that the proper noun is "AVA": https://github.com/avajs/ava/tree/main#how-is-the-name-written-and-pronounced

I updated to consistently use lowercase, with the (somewhat weak) rationale being reference to the npm package rather than the project.